### ISSUE TITLE
Provide nightly builds/updates

### ISSUE THREAD

**Issue by fraenki** [opnsense/core#371]
As a follow-up to #368 I suggest to provide nightly builds to make it easier to test bugfixes and new features. Maybe it is a good idea to provide two update channels "stable" and "nightly" (similar to FreeNAS).

**Comment by gitdevmod** [comment]
There is a development version https://forum.opnsense.org/index.php?topic=917.msg2640#msg2640

**Comment by fichtner** [comment]
I think most bits have been figured out on paper what it means to do nightly builds and how they differ from releases and how the mirrors are updated. BUT: it's still a lot of work to glue this together (securely in any way). We've set an initial target for 16.1, maybe materialising one or two months before the release in a way that it works and a switch in the GUI as well.

On a related note, the opnsense-devel package has been a huge milestone on the way to nightly development and stable builds. Thinking back, we've had no idea how to implement it. Just two weeks ago we came to realise that all our joined requirements can be implemented.

Note, however, that this only includes plugins and core packages, *not* ports and *definitely not* images. We'll have to do this step by step, avoiding pollution and ways to acquire a multitude of versions and flavours and images mixed together.
